Gutenboarding: add PlansGrid accordion

razvanpapadopol opened the pull request:
#### Changes proposed in this Pull Request
* Re-add plans accordion in Gutenboarding and Site setup. **Note: PlansGrid accordion was first introduced in #44299**
* Ship Feature picker and Plans Grid accordion to Gutenboarding "stable" (still behind `gutenboarding/feature-picker` flag in production environment).
* Update Gutenboarding e2e test.

#### Testing instructions
* Go to [/new](https://hash-88877d0524e87950b8e6138c974bc58e78661c17.calypso.live/new)
* Before reaching Plans step in the linear flow, you should land on Feature selection step.
* After skipping or confirming feature selection, Plans step should display as an accordion.
* After creating the site, apply D48381-code and sandbox the new site.
* When in editor Site setup flow should include PlansGrid accordion.

simison commented:
This still wouldn't launch it to customers, right? Either because we'd not be directing traffic to "experimental" flow yet or otherwise?

razvanpapadopol commented:
> This still wouldn't launch it to customers, right? Either because we'd not be directing traffic to "experimental" flow yet or otherwise?

Yes, the experimental branch of the flow would be hidden behind `?latest` query param for both [Gutenboarding site creation](https://github.com/Automattic/wp-calypso/pull/45054/files#diff-ccb31dc04d72d8e153a03c27c2475568R101) and [Complete Setup](https://github.com/Automattic/wp-calypso/pull/45054/files#diff-0662eb41df2f7435f27f7f0dd064a7e9R67) flow.

razvanpapadopol commented:
Superseded by #45280

simison commented:
@razvanpapadopol there's at least one E2E failure from Gutenboarding that needs fixing.

simison commented on client/landing/gutenboarding/stores/onboard/reducer.ts:
```diff
@@ -168,6 +155,19 @@ const selectedFeatures: Reducer< FeatureId[], OnboardAction > = (
 	return state;
 };
 
+const selectedFonts: Reducer< FontPair | undefined, OnboardAction > = (
```
These were alphabetized in https://github.com/Automattic/wp-calypso/pull/45145 — let's keep them tidy. :-) 

Also nicer to review this PR without unrelated changes.

simison commented on client/gutenberg/editor/calypsoify-iframe.tsx:
```diff
@@ -351,7 +351,7 @@ class CalypsoifyIframe extends Component<
 			const frankenflowUrl = `${ window.location.origin }/start/new-launch?siteSlug=${ this.props.siteSlug }&source=editor`;
 			const isNewLaunch = config.isEnabled( 'gutenboarding/new-launch' );
-			const isExperimental = new URLSearchParams( window?.location.search ).has( 'latest' );
+			const isExperimental = config.isEnabled( 'gutenboarding/feature-picker' );
```
This "isExperimental" = "feature-picker" can feel a bit confusing later on but I think it's fine now, and we'll sort it out later.

razvanpapadopol replied:
Right now this would be easiest way to flip the switch between multi-column PlansGrid and accordion PlansGrid + Feature picker for both Gutenboarding and Site Setup flows.

simison replied:
Makes sense 👍

simison reviewed (commented):
I'll test little more but two small visual findings. These can be follow-up PRs too:

Price loading state is a bit funky:
<img width="182" alt="Screenshot 2020-09-10 at 11 36 16" src="https://user-images.githubusercontent.com/87168/92703841-31205f00-f35b-11ea-9b0a-93955cdfd757.png">

Hiding the `/mo` completely vs placing it next to loading animation probably works best and is easiest.

Another is related to scroll positions:

- Expand all plans
- Click on domain to move to domains step
- Click continue
- Now you land in plans page where everything is collapsed again and scroll position isn't on top:

<img width="400" alt="Screenshot 2020-09-10 at 11 37 45" src="https://user-images.githubusercontent.com/87168/92704055-59a85900-f35b-11ea-9088-a8e1731137e1.png">
<img width="400" alt="Screenshot 2020-09-10 at 11 39 09" src="https://user-images.githubusercontent.com/87168/92704067-5b721c80-f35b-11ea-9b0c-ae2704d5f077.png">
<img width="400" alt="Screenshot 2020-09-10 at 11 39 13" src="https://user-images.githubusercontent.com/87168/92704079-5d3be000-f35b-11ea-972b-d251e9ef689f.png">

- Easiest would be at least to ensure we're on top when landing on plans page
- It might be helpful to store expanded/collapsed state during the session

Again, neither should block merging.

simison reviewed (commented):
In the launch flow, with flag enabled for a site that had free plan, I landed directly at plan step:

<img width="1586" alt="image" src="https://user-images.githubusercontent.com/87168/92712315-22897600-f362-11ea-9a83-d640039fe4ff.png">

I expected to land on domain step first like I do in `master`.

Without the flag, I see older "launch" button instead of "complete setup" button. I expected complete setup with multi-grid:

<img width="434" alt="image" src="https://user-images.githubusercontent.com/87168/92712413-4947ac80-f362-11ea-9d00-5526289ba5cc.png">

razvanpapadopol commented:
> Price loading state is a bit funky:

Fixed in https://github.com/Automattic/wp-calypso/pull/45054/commits/a659c2e8cb727a6dcabd52a6c97428cb30b3a055
Demo on mobile: https://cloudup.com/c4NHHotDhJ7
Demo on desktop: https://cloudup.com/cOd1LafAdQG

> Another is related to scroll positions:
> * Easiest would be at least to ensure we're on top when landing on plans page

Fixed in https://github.com/Automattic/wp-calypso/pull/45054/commits/88877d0524e87950b8e6138c974bc58e78661c17

simison reviewed (approved):
Double tested with new site with Chrome; looks like the landing on site title step was caused by something else than this PR (happens also in `master`).

Other than that, the flow is looking good and works great. 👍  Let's get this in.
